src/mmw/js/src/draw/views.js

```diff
@@ -131,11 +130,141 @@ function validatePointWithinDataSourceBounds(latlng, dataSource) {
     return d.promise();
 }
 
-var DrawWindow = Marionette.ItemView.extend({
+var DrawWindow = Marionette.LayoutView.extend({
     template: windowTmpl,
 
     id: 'draw-window',
 
+    regions: {
+        selectTypeRegion: '#select-area-region',
+        drawRegion: '#draw-region',
+        watershedDelineationRegion: '#place-marker-region'
+    },
+
+    ui: {
+        'selectBoundary': '#select-boundary-button',
+        'drawArea': '#draw-area-button',
+        'delineateWatershed': '#delineate-watershed-button',
+        'uploadFile': '#file-upload-button',
+        'resetDraw': '.reset-draw-button'
+    },
+
+    events: {
+        'click @ui.selectBoundary': 'toggleSelectBoundary',
+        'click @ui.drawArea': 'toggleDrawArea',
+        'click @ui.delineateWatershed': 'toggleDelineateWatershed',
+        'click @ui.uploadFile': 'toggleUploadFile',
+        'click @ui.resetDraw': 'resetDrawingState'
+    },
+
+    modelEvents: {
+        'change:selectedDrawTool': 'onShow'
+    },
+
+    constants: {
+        selectedDrawTool: 'selectedDrawTool',
+        selectBoundary: 'selectBoundary',
+        drawArea: 'drawArea',
+        delineateWatershed: 'delineateWatershed',
+        uploadFile: 'uploadFile'
+    },
+
+    templateHelpers: function() {
+        return {
+            activeDrawTool: this.model.get(this.constants.selectedDrawTool),
+        };
+    },
+
+    initialize: function() {
+        var map = App.getLeafletMap(),
+            ofg = L.featureGroup();
+        this.model.set('outlineFeatureGroup', ofg);
+        this.model.set(this.constants.selectedDrawTool, null);
+        map.addLayer(ofg);
+        this.rwdTaskModel = new models.RwdTaskModel();
+    },
+
+    onShow: function() {
+        var drawTools = settings.get('draw_tools'),
+            activeDrawTool = this.model.get(this.constants.selectedDrawTool);
+
+        // Update the view to mount the proper region divs
+        this.render();
```

Comment:
That makes sense -- I'll experiment with this to see if there's a better way to do it. The things that need to happen here are:

- update the active draw tool in the model
- add the region div for the active draw tool to the DOM
- call show on the region div to render the active draw tool

But it's likely this could get restructured to do that without calling `render` in `onShow`.